patches/api/0322-Small-StoneCutter-Events-API.patch

```diff
@@ -0,0 +1,91 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jayson <jaysonjson@web.de>
+Date: Thu, 24 Jun 2021 06:36:37 +0200
+Subject: [PATCH] Small StoneCutter Events -API
+
+
+diff --git a/src/main/java/io/papermc/paper/event/inventory/PrepareStonecuttingEvent.java b/src/main/java/io/papermc/paper/event/inventory/PrepareStonecuttingEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..1f0f2c9558dc2a774864ba2e43aa0af84d264174
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/inventory/PrepareStonecuttingEvent.java
+@@ -0,0 +1,43 @@
++package io.papermc.paper.event.inventory;
++
++import com.destroystokyo.paper.event.inventory.PrepareResultEvent;
++import org.bukkit.inventory.InventoryView;
++import org.bukkit.inventory.ItemStack;
++import org.bukkit.inventory.StonecutterInventory;
++import org.bukkit.inventory.StonecuttingRecipe;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++import java.util.List;
++
++/**
++ * Called when an item is put in a slot for cutting in a Stonecutter
++ */
++
++public class PrepareStonecuttingEvent extends PrepareResultEvent {
++
++    private final List<StonecuttingRecipe> recipes;
++    public PrepareStonecuttingEvent(@NotNull InventoryView inventory, @Nullable ItemStack result, List<StonecuttingRecipe> recipes) {
++        super(inventory, result);
++        this.recipes = recipes;
++    }
++
++
++    /**
++     * Recipe list from Input Item
++     * @return bukkit recipes
++     */
++    public List<StonecuttingRecipe> getRecipes() {
```

Comment:
Consider documenting the mutability of the returned List. Consider returning an unmodifiable/immutable list, since modifying the list does not actually modify the stonecutter recipes.